### PR TITLE
test: added test cases to change toggle based on channel

### PR DESCRIPTION
**Pull request by sundasnoreen12**
[INF-1243](https://2u-internal.atlassian.net/browse/INF-1243)

**Description**
Added following test cases for notification preference UI.

1. It updates all preferences in the column on web channel click when toggle state is true
2. It updates all preferences in the column on web channel click when toggle state is false

### CODE REVIEW

File: src/notification-preferences/ToggleSwitch.jsx [hunk 1]
```diff
@@ -7,12 +7,14 @@ const ToggleSwitch = ({
...
   disabled,
   onChange,
+  dataTestId,
 }) => (
   <Form.Switch
     name={name}
     checked={value}
     disabled={disabled}
     onChange={onChange}
+    data-testid={dataTestId}
```
Comment (awais-ansari) [inline src/notification-preferences/ToggleSwitch.jsx:17]: data-testid={id}

File: src/notification-preferences/ToggleSwitch.jsx [hunk 2]
```diff
@@ -21,11 +23,13 @@ ToggleSwitch.propTypes = {
   value: PropTypes.bool.isRequired,
   disabled: PropTypes.bool,
   onChange: PropTypes.func,
+  dataTestId: PropTypes.string,
```
Comment (awais-ansari) [inline src/notification-preferences/ToggleSwitch.jsx:26]: id: PropTypes.string,

File: src/notification-preferences/ToggleSwitch.jsx [hunk 3]
```diff
@@ -21,11 +23,13 @@ ToggleSwitch.propTypes = {
...
 };
 
 ToggleSwitch.defaultProps = {
   onChange: () => null,
   disabled: false,
+  dataTestId: '',
```
Comment (awais-ansari) [inline src/notification-preferences/ToggleSwitch.jsx:32]: id: '',

File: src/notification-preferences/NotificationPreferenceRow.jsx [hunk 4]
```diff
@@ -78,6 +78,7 @@ const NotificationPreferenceRow = ({ appId, preferenceName }) => {
               value={preference[channel]}
               onChange={onToggle}
               disabled={nonEditable.includes(channel) || updatePreferencesStatus === LOADING_STATUS}
+              dataTestId={`${preferenceName}-${channel}`}
```
Comment (awais-ansari) [inline src/notification-preferences/NotificationPreferenceRow.jsx:81]: id = {`${preferenceName}-${channel}`}

File: src/notification-preferences/NotificationPreferences.test.jsx [hunk 5]
```diff
@@ -78,17 +97,19 @@ const setupStore = (override = {}) => {
   return store;
 };
 
-const renderComponent = (store = {}) => render(
+const renderComponent = (store = {}) => (
```
Comment (awais-ansari) [inline src/notification-preferences/NotificationPreferences.test.jsx:100]: renderComponent  => notificationPreferences

File: src/notification-preferences/NotificationPreferences.test.jsx [hunk 6]
```diff
@@ -33,33 +35,50 @@ const defaultPreferences = {
...
+    },
+    {
+      id: 'newComment', appId: 'discussion', web: toggleVal, push: toggleVal, email: toggleVal,
+    },
+    {
+      id: 'newAssignment', appId: 'coursework', web: toggleVal, push: toggleVal, email: toggleVal,
+    },
+    {
+      id: 'newGrade', appId: 'coursework', web: toggleVal, push: toggleVal, email: toggleVal,
+    },
```
Comment (awais-ansari) [inline src/notification-preferences/NotificationPreferences.test.jsx:78]: {id: 'newPost', appId: 'discussion', web: toggleVal},
 {id: 'newComment', appId: 'discussion', web: toggleVal},
 {id: 'newAssignment', appId: 'coursework', web: toggleVal},

File: src/notification-preferences/NotificationPreferences.test.jsx [hunk 7]
```diff
@@ -139,7 +161,43 @@ describe('Notification Preferences', () => {
...
     expect(screen.queryByTestId('not-found-page')).toBeInTheDocument();
   });
+
+  it.each([false, true])(
+    'updates all preferences in the column on web channel click when toggle state is - %s',
+    async (toggleState) => {
+      store = setupStore(updateChannelPreferences(toggleState));
+      const wrapper = render(renderComponent(store));
+
+      const getCheckbox = (id) => screen.queryByTestId(`${id}-web`);
```
Comment (awais-ansari) [inline src/notification-preferences/NotificationPreferences.test.jsx:174]: getCheckbox  => getChannelSwitch

File: src/notification-preferences/NotificationPreferences.test.jsx [hunk 8]
```diff
@@ -139,7 +161,43 @@ describe('Notification Preferences', () => {
...
+      const checkboxes = ['newPost', 'newComment', 'newAssignment', 'newGrade'];
```
Comment (awais-ansari) [inline src/notification-preferences/NotificationPreferences.test.jsx:175]: notificationTypes

**Review by awais-ansari — APPROVED** [review]

**Review by ayesha-waris — APPROVED** [review]

**Pull request merged**